docs/write_tests.rst

```diff
@@ -0,0 +1,131 @@
+
+In the code below, we have our ``TestMailchimp`` that serves as our test case. We have one test method to test calls to our ``get_campaigns`` method in our ``Mailchimp`` Connector class.
+
+We decorate ``test_get_campaigns`` with the `requests_mock.Mocker`` class, which passes in our requests mocker as an argument ``m``. We can then use this mock to configure our mock server.
+
+To configure our mock API, we can make use of the ``get``, ``post``, ``patch``, ``put``, and ``delete`` methods on the mock object ``m``. These methods mirror the calls we expect our Connector to make to the API. The methods on the mock object correspond to the HTTP method that we are expecting our Connector class to call with, and the first argument to the mock method is the URL that we expect our Connector to call. We can then provide a ``json`` argument to the mock method to provide a canned response that the mock HTTP server should return.
+
+Below, we will use the ``get`` method to tell the mock HTTP server that we expect a call to the ``campaigns`` endpoint of our API, and that the mock HTTP server should return our ``test_campaignss` canned response. The Mailchimp tests store the canned responses in a separate Python module, which can help keep the test code separate from the test data, which helps makes tests more readable.
+
+.. code-block:: python
+
+    class TestMailchimp(unittest.TestCase):
+
+        @requests_mock.Mocker()
+        def test_get_campaigns(self, m):
+
+            mc = MailChimp(api_key='12345')
+
+            # Test that campaigns are returned correctly.
+            m.get(self.mc.uri + 'campaigns', json=expected_json.test_campaigns)
+            tbl = self.mc.get_campaigns()
+
+            self.assertEqual(tbl.num_rows, 2)
+
+
+After wiring up the mock HTTP API, we are ready to call the ``get_campaigns`` method on our ``Mailchimp`` connector. Since we are using the ``requests_mock.Mocker`` class, our Connector will not actually hit the Mailchimp API; our call will be intercepted and passed to our mock HTTP server.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Tests for Connectors Built on Third Party Libraries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For classes that wrap third party libraries, we are looking to simulate whatever client object that serves as the Python API for the calls to the external service. To simulate the client object, we can use the ``unittest.mock.MagicMock`` class from the Python standard library.
+
+The ``MagicMock`` class allows us to build a fake version of the client object by simulating functionality we expect our Connector class to use -- namely by returning canned responses to method calls on the client. As well, the ``MagicMock`` class is recording those calls as they happen, so that after we have run through our tests, we can check that things happened in the way we expected.
+
+The following gives a simple example of initializing, setting up, and using a ``MagicMock`` class:
```

Comment:
This might lead to some confusion about "using" the mock, since the example code here is focused on setting up and checking expectations. Calling the mock directly like this is just for illustrative value.

```suggestion
The following gives a simple example of initializing, setting up, and checking expectations on a ``MagicMock`` class. Please see the following, SalesforceTest example on integrating it with your Connector class.
```